src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -38,8 +40,11 @@ public IntegrityCheck(BibDatabaseContext bibDatabaseContext) {
             return result;
         }
 
-        result.addAll(new AuthorNameChecker("author").check(entry));
-        result.addAll(new AuthorNameChecker("editor").check(entry));
+        for (String fieldName : entry.getFieldNames()) {
+            if (InternalBibtexFields.getFieldExtras(fieldName).contains(BibtexSingleFieldProperties.PERSON_NAMES)) {
```

Comment:
I think performance is no issue here as you iterate over like 10 items, so I would put my priority on readability. On second thought, I would even propose to move the code to AuthorNameChecker and remove its constructor argument.